tests/plugins/test_ddp_spawn_plugin.py

```diff
@@ -40,3 +52,41 @@ def test_ddp_cpu():
     model = BoringModelDDPCPU()
 
     trainer.fit(model)
+
+
+@RunIf(min_gpus=1)
+def test_ddp_gpu():
+    """Tests if device is set correctely when training for DDPSpawnPlugin."""
+    trainer = Trainer(num_processes=2, fast_dev_run=True, gpus=1, accelerator='ddp_spawn')
+
+    assert isinstance(trainer.training_type_plugin, DDPSpawnPlugin)
+    assert trainer.training_type_plugin.on_gpu
+    assert trainer.training_type_plugin.root_device == torch.device("cuda:0")
+
+    val: float = 1.0
+    val_name: str = "val_acc"
+    model = BoringCallbackDDPSpawnModel(val_name, val)
+    dm = BoringDataModule()
+
+    trainer.fit(model, datamodule=dm)
+    assert "callback_metrics" in trainer.spawn_extra_parameters
+    assert trainer.spawn_extra_parameters["callback_metrics"][val_name] == val
+
+
+@RunIf(min_gpus=2)
+def test_ddp_multigpu():
```

Comment:
this test is identical to the above with the exception of the gpu setting. 
Can we just do the 2 gpu case? I don't see the value for single gpu test